### PR TITLE
feat(exposition)!: add `content-meta` header

### PR DESCRIPTION
**Pull request by temich**
Allow to pass initial `meta` value using non-standard `content-meta` request header.

```http
POST /files/ HTTP/1.1
content-type: application/octet-stream
content-meta: foo, bar=baz
```

```yaml
meta:
  foo: 'true'
  bar: baz
```

**BREAKING CHANGES**

- `prefix` option for Temporary Storage provider is renamed to `directory` (since it's not used as a prefix anymore)
- `@toa.io/http` is renamed to `@toa.io/agent`

### CODE REVIEW

File: extensions/exposition/source/directives/octets/Store.ts [hunk 1]
```diff
@@ -141,12 +152,29 @@ export class Store implements Directive {
...
   }
+
+  private parseMeta (value: string | string[]): Record<string, string> {
+    if (Array.isArray(value))
+      value = value.join(',')
+
+    const meta: Record<string, string> = {}
+
+    for (const pair of value.split(',')) {
+      const [k, v] = pair.trim().split('=')
```
Comment (tinovyatkin) [inline extensions/exposition/source/directives/octets/Store.ts]: Let's add `limit` parameter to `.split` to allow `=` in values (i.e. base64 encoded things)

```suggestion
      const [k, v] = pair.trim().split('=', 2)
```

File: extensions/exposition/source/directives/octets/Store.ts [hunk 2]
```diff
@@ -141,12 +152,29 @@ export class Store implements Directive {
 
     return await this.discovery[key]
   }
+
+  private parseMeta (value: string | string[]): Record<string, string> {
+    if (Array.isArray(value))
+      value = value.join(',')
+
+    const meta: Record<string, string> = {}
```
Comment (tinovyatkin) [inline extensions/exposition/source/directives/octets/Store.ts:160]: NIT: I'm seeing this pattern (creating empty object and then populating it's properties) very frequently in Toa codebase - is there a reason why we avoiding `Object.fromEntries(something.map(...)))` ?

File: extensions/storages/schemas/fs.cos.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+type: object
+properties:
+  provider:
+    const: fs
+  path:
+    type: string
```
Comment (tinovyatkin) [inline extensions/storages/schemas/fs.cos.yaml:6]: NIT: This probably may have `maxLength` limit also

File: extensions/storages/schemas/s3.cos.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  provider:
+    const: s3
+  bucket:
+    type: string
+  region:
+    type: string
+  prefix:
+    type: string
+  endpoint:
+    type: string
```
Comment (tinovyatkin) [inline extensions/storages/schemas/s3.cos.yaml:14]: this should be `format: uri` probably?

File: extensions/storages/schemas/s3.cos.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+type: object
+properties:
+  provider:
+    const: s3
+  bucket:
+    type: string
```
Comment (tinovyatkin) [inline extensions/storages/schemas/s3.cos.yaml:6]: This should have way more validations: https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html#general-purpose-bucket-names

File: extensions/exposition/source/directives/octets/Store.ts (hunk 1 above)
Comment (temich) [reply extensions/exposition/source/directives/octets/Store.ts]: Updated (`split` doesn't work this way).

File: extensions/exposition/source/directives/octets/Store.ts (hunk 2 above)
Comment (temich) [reply extensions/exposition/source/directives/octets/Store.ts:160]: Not sure, just seems more readable to me.

Why `fromEntries` is better? In terms of performance and GC, it's not clear to me which is worse: update the object or create a bunch of arrays.

File: extensions/storages/schemas/fs.cos.yaml (hunk 3 above)
Comment (temich) [reply extensions/storages/schemas/fs.cos.yaml:6]: Done

File: extensions/storages/schemas/s3.cos.yaml (hunk 4 above)
Comment (temich) [reply extensions/storages/schemas/s3.cos.yaml:14]: Done

File: extensions/storages/schemas/s3.cos.yaml (hunk 5 above)
Comment (temich) [reply extensions/storages/schemas/s3.cos.yaml:6]: I believe this is not worth the effort.

In the worst-case scenario, a developer may experience a runtime exception from the AWS SDK.

Comment (temich) [reply extensions/storages/schemas/s3.cos.yaml:6]: Yet, I've added length restrictions.

File: extensions/exposition/source/directives/octets/Store.ts (hunk 1 above)
Comment (tinovyatkin) [reply extensions/exposition/source/directives/octets/Store.ts]: NIT: yeah, something like that would do:
```ts
const [k, v] = pair.trim().split(/=(.*)/s, 2)
```

**Review by tinovyatkin — APPROVED** [review]

File: extensions/exposition/source/directives/octets/Store.ts (hunk 2 above)
Comment (tinovyatkin) [reply extensions/exposition/source/directives/octets/Store.ts:160]: It's definitely not better from V8 performance point of view (as many other higher level operations like `for(const a of iterable)` are still slower than `for(let i=0; i++; i<`) but it's more idiomatic tool of the language as what you actually want to here is to create an object from entries and [Object.fromEntries](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/fromEntries) is the way the language runtime provides for that.

Comment (temich) [reply extensions/exposition/source/directives/octets/Store.ts:160]: Each time I write JavaScript that creates an object it hurts me a little bit. Constructing an array of tuples within a loop is simply a lot of pain.

**Pull request merged**